# [Rosalind](http://rosalind.info/problems/list-view/)
Rosalind is a platform for learning bioinformatics and programming through problem solving.

## Problems (285 in total @20181125)
Python 3.5+ is recommended until now. It's the future.

### [PV - Python Village](http://rosalind.info/problems/list-view/?location=python-village)
If you are completely new to programming, try these initial problems to learn a few basics about the Python programming language. You'll get familiar with the operations needed to start solving bioinformatics challenges in the Stronghold.

6 Problems:
[INI1](http://rosalind.info/problems/ini1/) [`Installing Python`](https://github.com/lizc07/rosalind/blob/master/PV_INI1.py);
[INI2](http://rosalind.info/problems/ini2/) [`Variables and Some Arithmetic`](https://github.com/lizc07/rosalind/blob/master/PV_INI2.py);
[INI3](http://rosalind.info/problems/ini3/) [`Strings and Lists`](https://github.com/lizc07/rosalind/blob/master/PV_INI3.py);
[INI4](http://rosalind.info/problems/ini4/) [`Conditions and Loops`](https://github.com/lizc07/rosalind/blob/master/PV_INI4.py);
[INI5](http://rosalind.info/problems/ini5/) [`Working with Files`](https://github.com/lizc07/rosalind/blob/master/PV_INI5.py);
[INI6](http://rosalind.info/problems/ini6/) [`Dictionaries`](https://github.com/lizc07/rosalind/blob/master/PV_INI6.py);

### [BS - Bioinformatics Stronghold](http://rosalind.info/problems/list-view/)
Discover the algorithms underlying a variety of bioinformatics topics: computational mass spectrometry, alignment, dynamic programming, genome assembly, genome rearrangements, phylogeny, probability, string algorithms and others.

105 Problemsï¼š
[DNA](http://rosalind.info/problems/dna/) [`Counting DNA Nucleotides`](https://github.com/lizc07/rosalind/blob/master/BS_DNA.py);
[RNA](http://rosalind.info/problems/rna/) [`Transcribing DNA into RNA`](https://github.com/lizc07/rosalind/blob/master/BS_RNA.py);
[REVC](http://rosalind.info/problems/revc/) [`Complementing a Strand of DNA`](https://github.com/lizc07/rosalind/blob/master/BS_REVC.py);
[FIB](http://rosalind.info/problems/fib/) [`Rabbits and Recurrence Relations`](https://github.com/lizc07/rosalind/blob/master/BS_FIB.py);
[GC](http://rosalind.info/problems/gc/) [`Computing GC Content`](https://github.com/lizc07/rosalind/blob/master/BS_GC.py);
[HAMM](http://rosalind.info/problems/hamm/) [`Counting Point Mutations`](https://github.com/lizc07/rosalind/blob/master/BS_HAMM.py);
[IPRB](http://rosalind.info/problems/iprb/) [`Mendel's First Law`](https://github.com/lizc07/rosalind/blob/master/BS_IPRB.py);
[PROT](http://rosalind.info/problems/prot/) [`Translating RNA into Protein`](https://github.com/lizc07/rosalind/blob/master/BS_PROT.py);
[SUBS](http://rosalind.info/problems/subs/) [`Finding a Motif in DNA`](https://github.com/lizc07/rosalind/blob/master/BS_SUBS.py);
[CONS](http://rosalind.info/problems/cons/) [`Consensus and Profile`](https://github.com/lizc07/rosalind/blob/master/BS_CONS.py);
[FIBD](http://rosalind.info/problems/fibd/) [`Mortal Fibonacci Rabbits`](https://github.com/lizc07/rosalind/blob/master/BS_FIBD.py);
[GRPH](http://rosalind.info/problems/grph/) [`Overlap Graphs`](https://github.com/lizc07/rosalind/blob/master/BS_GRPH.py);
[IEV](http://rosalind.info/problems/iev/) [`Calculating Expected Offspring`](https://github.com/lizc07/rosalind/blob/master/BS_IEV.py);
[LCSM](http://rosalind.info/problems/lcsm/) [`Finding a Shared Motif`](https://github.com/lizc07/rosalind/blob/master/BS_LCSM.py);
[LIA](http://rosalind.info/problems/lia/) [`Independent Alleles`](https://github.com/lizc07/rosalind/blob/master/BS_LIA.py);
[MPRT](http://rosalind.info/problems/mprt/) [`Finding a Protein Motif`](https://github.com/lizc07/rosalind/blob/master/BS_MPRT.py);
[MRNA](http://rosalind.info/problems/mrna/) [`Inferring mRNA from Protein`](https://github.com/lizc07/rosalind/blob/master/BS_MRNA.py);
[ORF](http://rosalind.info/problems/orf/) [`Open Reading Frames`](https://github.com/lizc07/rosalind/blob/master/BS_ORF.py);
[PERM](http://rosalind.info/problems/perm/) [`Enumerating Gene Orders`](https://github.com/lizc07/rosalind/blob/master/BS_PERM.py);
[PRTM](http://rosalind.info/problems/prtm/) [`Calculating Protein Mass`](https://github.com/lizc07/rosalind/blob/master/BS_PRTM.py);
[REVP](http://rosalind.info/problems/revp/) [`Locating Restriction Sites`](https://github.com/lizc07/rosalind/blob/master/BS_REVP.py);
[SPLC](http://rosalind.info/problems/splc/) [`RNA Splicing`](https://github.com/lizc07/rosalind/blob/master/BS_SPLC.py);
[LEXF](http://rosalind.info/problems/lexf/) [`Enumerating k-mers Lexicographically`](https://github.com/lizc07/rosalind/blob/master/BS_LEXF.py);
[LGIS](http://rosalind.info/problems/lgis/) [`Longest Increasing Subsequence`](https://github.com/lizc07/rosalind/blob/master/BS_LGIS.py);
[LONG](http://rosalind.info/problems/long/) [`Genome Assembly as Shortest Superstring`](https://github.com/lizc07/rosalind/blob/master/BS_LONG.py);
[PMCH](http://rosalind.info/problems/pmch/) [`Perfect Matchings and RNA Secondary Structures`](https://github.com/lizc07/rosalind/blob/master/BS_PMCH.py);
[PPER](http://rosalind.info/problems/pper/) [`Partial Permutations`](https://github.com/lizc07/rosalind/blob/master/BS_PPER.py);
[PROB](http://rosalind.info/problems/prob/) [`Introduction to Random Strings`](https://github.com/lizc07/rosalind/blob/master/BS_PROB.py);
[SIGN](http://rosalind.info/problems/sign/) [`Enumerating Oriented Gene Orderings`](https://github.com/lizc07/rosalind/blob/master/BS_SIGN.py);
[SSEQ](http://rosalind.info/problems/sseq/) [`Finding a Spliced Motif`](https://github.com/lizc07/rosalind/blob/master/BS_SSEQ.py);
[TRAN](http://rosalind.info/problems/tran/) [`Transitions and Transversions`](https://github.com/lizc07/rosalind/blob/master/BS_TRAN.py);
[TREE](http://rosalind.info/problems/tree/) [`Completing a Tree`](https://github.com/lizc07/rosalind/blob/master/BS_TREE.py);
[CAT](http://rosalind.info/problems/cat/) [`Catalan Numbers and RNA Secondary Structures`](https://github.com/lizc07/rosalind/blob/master/BS_CAT.py);
[CORR](http://rosalind.info/problems/corr/) [`Error Correction in Reads`](https://github.com/lizc07/rosalind/blob/master/BS_CORR.py);
[INOD](http://rosalind.info/problems/inod/) [`Counting Phylogenetic Ancestors`](https://github.com/lizc07/rosalind/blob/master/BS_INOD.py);
[KMER](http://rosalind.info/problems/kmer/) [`k-Mer Composition`](https://github.com/lizc07/rosalind/blob/master/BS_KMER.py);
[KMP](http://rosalind.info/problems/kmp/) [`Speeding Up Motif Finding`](https://github.com/lizc07/rosalind/blob/master/BS_KMP.py);
[LCSQ](http://rosalind.info/problems/lcsq/) [`Finding a Shared Spliced Motif`](https://github.com/lizc07/rosalind/blob/master/BS_LCSQ.py);
[LEXV](http://rosalind.info/problems/lexv/) [`Ordering Strings of Varying Length Lexicographically`](https://github.com/lizc07/rosalind/blob/master/BS_LEXV.py);
[MMCH](http://rosalind.info/problems/mmch/) [`Maximum Matchings and RNA Secondary Structures`](https://github.com/lizc07/rosalind/blob/master/BS_MMCH.py);
[PDST](http://rosalind.info/problems/pdst/) [`Creating a Distance Matrix`](https://github.com/lizc07/rosalind/blob/master/BS_PDST.py);
[REAR](http://rosalind.info/problems/rear/) [`Reversal Distance`](https://github.com/lizc07/rosalind/blob/master/BS_REAR.py);
[RSTR](http://rosalind.info/problems/rstr/) [`Matching Random Motifs`](https://github.com/lizc07/rosalind/blob/master/BS_RSTR.py);
[SSET](http://rosalind.info/problems/sset/) [`Counting Subsets`](https://github.com/lizc07/rosalind/blob/master/BS_SSET.py);
[ASPC](http://rosalind.info/problems/aspc/) [`Introduction to Alternative Splicing`](https://github.com/lizc07/rosalind/blob/master/BS_ASPC.py);
[EDIT](http://rosalind.info/problems/edit/) [`Edit Distance`](https://github.com/lizc07/rosalind/blob/master/BS_EDIT.py);
[EVAL](http://rosalind.info/problems/eval/) [`Expected Number of Restriction Sites`](https://github.com/lizc07/rosalind/blob/master/BS_EVAL.py);
[MOTZ](http://rosalind.info/problems/motz/) [`Motzkin Numbers and RNA Secondary Structures`](https://github.com/lizc07/rosalind/blob/master/BS_MOTZ.py);
[NWCK](http://rosalind.info/problems/nwck/) [`Distances in Trees`](https://github.com/lizc07/rosalind/blob/master/BS_NWCK.py);
[SCSP](http://rosalind.info/problems/scsp/) [`Interleaving Two Motifs`](https://github.com/lizc07/rosalind/blob/master/BS_SCSP.py);
[SETO](http://rosalind.info/problems/seto/) [`Introduction to Set Operations`](https://github.com/lizc07/rosalind/blob/master/BS_SETO.py);
[SORT](http://rosalind.info/problems/sort/) [`Sorting by Reversals`](https://github.com/lizc07/rosalind/blob/master/BS_SORT.py);
[SPEC](http://rosalind.info/problems/spec/) [`Inferring Protein from Spectrum`](https://github.com/lizc07/rosalind/blob/master/BS_SPEC.py);
[TRIE](http://rosalind.info/problems/trie/) [`Introduction to Pattern Matching`](https://github.com/lizc07/rosalind/blob/master/BS_TRIE.py);
[CONV](http://rosalind.info/problems/conv/) [`Comparing Spectra with the Spectral Convolution`](https://github.com/lizc07/rosalind/blob/master/BS_CONV.py);
[CTBL](http://rosalind.info/problems/ctbl/) [`Creating a Character Table`](https://github.com/lizc07/rosalind/blob/master/BS_CTBL.py);
[DBRU](http://rosalind.info/problems/dbru/) [`Constructing a De Bruijn Graph`](https://github.com/lizc07/rosalind/blob/master/BS_DBRU.py);
[EDTA](http://rosalind.info/problems/edta/) [`Edit Distance Alignment`](https://github.com/lizc07/rosalind/blob/master/BS_EDTA.py);
[FULL](http://rosalind.info/problems/full/) [`Inferring Peptide from Full Spectrum`](https://github.com/lizc07/rosalind/blob/master/BS_FULL.py);
[INDC](http://rosalind.info/problems/indc/) [`Independent Segregation of Chromosomes`](https://github.com/lizc07/rosalind/blob/master/BS_INDC.py);
[ITWV](http://rosalind.info/problems/itwv/) [`Finding Disjoint Motifs in a Gene`](https://github.com/lizc07/rosalind/blob/master/BS_ITWV.py);
[LREP](http://rosalind.info/problems/lrep/) [`Finding the Longest Multiple Repeat`](https://github.com/lizc07/rosalind/blob/master/BS_LREP.py);
[NKEW](http://rosalind.info/problems/nkew/) [`Newick Format with Edge Weights`](https://github.com/lizc07/rosalind/blob/master/BS_NKEW.py);
[RNAS](http://rosalind.info/problems/rnas/) [`Wobble Bonding and RNA Secondary Structures`](https://github.com/lizc07/rosalind/blob/master/BS_RNAS.py);
[AFRQ](http://rosalind.info/problems/afrq/) [`Counting Disease Carriers`](https://github.com/lizc07/rosalind/blob/master/BS_AFRQ.py);
[CSTR](http://rosalind.info/problems/cstr/) [`Creating a Character Table from Genetic Strings`](https://github.com/lizc07/rosalind/blob/master/BS_CSTR.py);
[CTEA](http://rosalind.info/problems/ctea/) [`Counting Optimal Alignments`](https://github.com/lizc07/rosalind/blob/master/BS_CTEA.py);
[CUNR](http://rosalind.info/problems/cunr/) [`Counting Unrooted Binary Trees`](https://github.com/lizc07/rosalind/blob/master/BS_CUNR.py);
[GLOB](http://rosalind.info/problems/glob/) [`Global Alignment with Scoring Matrix`](https://github.com/lizc07/rosalind/blob/master/BS_GLOB.py);
[PCOV](http://rosalind.info/problems/pcov/) [`Genome Assembly with Perfect Coverage`](https://github.com/lizc07/rosalind/blob/master/BS_PCOV.py);
[PRSM](http://rosalind.info/problems/prsm/) [`Matching a Spectrum to a Protein`](https://github.com/lizc07/rosalind/blob/master/BS_PRSM.py);
[QRT](http://rosalind.info/problems/qrt/) [`Quartets`](https://github.com/lizc07/rosalind/blob/master/BS_QRT.py);
[SGRA](http://rosalind.info/problems/sgra/) [`Using the Spectrum Graph to Infer Peptides`](https://github.com/lizc07/rosalind/blob/master/BS_SGRA.py);
[SUFF](http://rosalind.info/problems/suff/) [`Encoding Suffix Trees`](https://github.com/lizc07/rosalind/blob/master/BS_SUFF.py);
[CHBP](http://rosalind.info/problems/chbp/) [`Character-Based Phylogeny`](https://github.com/lizc07/rosalind/blob/master/BS_CHBP.py);
[CNTQ](http://rosalind.info/problems/cntq/) [`Counting Quartets`](https://github.com/lizc07/rosalind/blob/master/BS_CNTQ.py);
[EUBT](http://rosalind.info/problems/eubt/) [`Enumerating Unrooted Binary Trees`](https://github.com/lizc07/rosalind/blob/master/BS_EUBT.py);
[GASM](http://rosalind.info/problems/gasm/) [`Genome Assembly Using Reads`](https://github.com/lizc07/rosalind/blob/master/BS_GASM.py);
[GCON](http://rosalind.info/problems/gcon/) [`Global Alignment with Constant Gap Penalty`](https://github.com/lizc07/rosalind/blob/master/BS_GCON.py);
[LING](http://rosalind.info/problems/ling/) [`Linguistic Complexity of a Genome`](https://github.com/lizc07/rosalind/blob/master/BS_LING.py);
[LOCA](http://rosalind.info/problems/loca/) [`Local Alignment with Scoring Matrix`](https://github.com/lizc07/rosalind/blob/master/BS_LOCA.py);
[MEND](http://rosalind.info/problems/mend/) [`Inferring Genotype from a Pedigree`](https://github.com/lizc07/rosalind/blob/master/BS_MEND.py);
[MGAP](http://rosalind.info/problems/mgap/) [`Maximizing the Gap Symbols of an Optimal Alignment`](https://github.com/lizc07/rosalind/blob/master/BS_MGAP.py);
[MREP](http://rosalind.info/problems/mrep/) [`Identifying Maximal Repeats`](https://github.com/lizc07/rosalind/blob/master/BS_MREP.py);
[MULT](http://rosalind.info/problems/mult/) [`Multiple Alignment`](https://github.com/lizc07/rosalind/blob/master/BS_MULT.py);
[PDPL](http://rosalind.info/problems/pdpl/) [`Creating a Restriction Map`](https://github.com/lizc07/rosalind/blob/master/BS_PDPL.py);
[ROOT](http://rosalind.info/problems/root/) [`Counting Rooted Binary Trees`](https://github.com/lizc07/rosalind/blob/master/BS_ROOT.py);
[SEXL](http://rosalind.info/problems/sexl/) [`Sex-Linked Inheritance`](https://github.com/lizc07/rosalind/blob/master/BS_SEXL.py);
[SPTD](http://rosalind.info/problems/sptd/) [`Phylogeny Comparison with Split Distance`](https://github.com/lizc07/rosalind/blob/master/BS_SPTD.py);
[WFMD](http://rosalind.info/problems/wfmd/) [`The Wright-Fisher Model of Genetic Drift`](https://github.com/lizc07/rosalind/blob/master/BS_WFMD.py);
[ALPH](http://rosalind.info/problems/alph/) [`Alignment-Based Phylogeny`](https://github.com/lizc07/rosalind/blob/master/BS_ALPH.py);
[ASMQ](http://rosalind.info/problems/asmq/) [`Assessing Assembly Quality with N50 and N75`](https://github.com/lizc07/rosalind/blob/master/BS_ASMQ.py);
[CSET](http://rosalind.info/problems/cset/) [`Fixing an Inconsistent Character Set`](https://github.com/lizc07/rosalind/blob/master/BS_CSET.py);
[EBIN](http://rosalind.info/problems/ebin/) [`Wright-Fisher's Expected Behavior`](https://github.com/lizc07/rosalind/blob/master/BS_EBIN.py);
[FOUN](http://rosalind.info/problems/foun/) [`The Founder Effect and Genetic Drift`](https://github.com/lizc07/rosalind/blob/master/BS_FOUN.py);
[GAFF](http://rosalind.info/problems/gaff/) [`Global Alignment with Scoring Matrix and Affine Gap Penalty`](https://github.com/lizc07/rosalind/blob/master/BS_GAFF.py);
[GREP](http://rosalind.info/problems/grep/) [`Genome Assembly with Perfect Coverage and Repeats`](https://github.com/lizc07/rosalind/blob/master/BS_GREP.py);
[OAP](http://rosalind.info/problems/oap/) [`Overlap Alignment`](https://github.com/lizc07/rosalind/blob/master/BS_OAP.py);
[QRTD](http://rosalind.info/problems/qrtd/) [`Quartet Distance`](https://github.com/lizc07/rosalind/blob/master/BS_QRTD.py);
[SIMS](http://rosalind.info/problems/sims/) [`Finding a Motif with Modifications`](https://github.com/lizc07/rosalind/blob/master/BS_SIMS.py);
[SMGB](http://rosalind.info/problems/smgb/) [`Semiglobal Alignment`](https://github.com/lizc07/rosalind/blob/master/BS_SMGB.py);
[KSIM](http://rosalind.info/problems/ksim/) [`Finding All Similar Motifs`](https://github.com/lizc07/rosalind/blob/master/BS_KSIM.py);
[LAFF](http://rosalind.info/problems/laff/) [`Local Alignment with Affine Gap Penalty`](https://github.com/lizc07/rosalind/blob/master/BS_LAFF.py);
[OSYM](http://rosalind.info/problems/osym/) [`Isolating Symbols in Alignments`](https://github.com/lizc07/rosalind/blob/master/BS_OSYM.py);
[RSUB](http://rosalind.info/problems/rsub/) [`Identifying Reversing Substitutions`](https://github.com/lizc07/rosalind/blob/master/BS_RSUB.py);

### [BA - Bioinformatics Armory](http://rosalind.info/problems/list-view/?location=bioinformatics-armory)
Ready-to-use software tools abound for bioinformatics analysis. Whereas in the Stronghold you implement algorithms on your own, in the Armory you solve similar problems by using existing tools.
### [BT - Bioinformatics Textbook Track](http://rosalind.info/problems/list-view/?location=bioinformatics-textbook-track)
A collection of exercises to accompany Bioinformatics Algorithms: An Active-Learning Approach by Phillip Compeau & Pavel Pevzner. A full version of this text is hosted on stepic.org
### [AH - Algorithmic Heights](http://rosalind.info/problems/list-view/?location=algorithmic-heights)
A collection of exercises in introductory algorithms to accompany "Algorithms", the popular textbook by Dasgupta, Papadimitriou, and Vazirani.

------------------------------
